packages/rum-core/src/domain/longTask/longTaskCollection.spec.ts

```diff
@@ -1,29 +1,38 @@
 import type { RelativeTime, ServerDuration } from '@datadog/browser-core'
-import type { RumSessionManagerMock, TestSetupBuilder } from '../../../test'
-import { createPerformanceEntry, createRumSessionManagerMock, mockPerformanceObserver, setup } from '../../../test'
+import { createPerformanceEntry, mockPerformanceObserver, validateRumEventFormat } from '../../../test'
 import { RumPerformanceEntryType } from '../../browser/performanceObservable'
+import type { RawRumEvent } from '../../rawRumEvent.types'
 import { RumEventType } from '../../rawRumEvent.types'
-import { LifeCycleEventType } from '../lifeCycle'
+import type { RawRumEventCollectedData } from '../lifeCycle'
+import { LifeCycle, LifeCycleEventType } from '../lifeCycle'
+import type { RumConfiguration } from '../configuration'
 import { startLongTaskCollection } from './longTaskCollection'
 
 describe('long task collection', () => {
-  let setupBuilder: TestSetupBuilder
-  let sessionManager: RumSessionManagerMock
-  let trackLongTasks: boolean
+  let lifeCycle = new LifeCycle()
+  let rawRumEvents: Array<RawRumEventCollectedData<RawRumEvent>> = []
+  let stopLongTaskCollection: () => void
 
-  beforeEach(() => {
-    trackLongTasks = true
-    sessionManager = createRumSessionManagerMock()
-    setupBuilder = setup()
-      .withSessionManager(sessionManager)
-      .beforeBuild(({ lifeCycle, configuration }) => {
-        startLongTaskCollection(lifeCycle, { ...configuration, trackLongTasks })
-      })
+  function setupLongTaskCollection(trackLongTasks = true) {
+    lifeCycle = new LifeCycle()
+    startLongTaskCollection(lifeCycle, { trackLongTasks } as RumConfiguration)
+
+    const eventsSubscription = lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
+      rawRumEvents.push(data)
+      validateRumEventFormat(data.rawRumEvent)
+    })
```

Comment:
**💬 suggestion:** this pattern is used 6 times in specs, you could ‏factorize this with something like (maybe in rum-core/test-formatValidation.ts):

```ts
function collectAndValidateRawRumEvents(lifeCycle) {
  let rawRumEvents = []
  lifeCycle.subscribe(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, (data) => {
    rawRumEvents.push(data)
    validateRumEventFormat(data.rawRumEvent)
  })
  registerCleanupTask(() => {
    rawRumEvents = [] // this might not be necessary
  })
  return rawRumEvents
}
```